src/CommonITILObject.php

```diff
@@ -8139,7 +8139,25 @@ protected function updateActors()
                     }
                 }
 
-                if ($found === false) {
+
+                if ($existing['itemtype'] === 'User') {
+                    $input_field_name = '_additional_' . $actortype_str . 's';
+                } else {
+                    $type = strtolower($existing['itemtype']::getType());
+                    $input_field_name = '_additional_' . $type . 's_' . $actortype_str . 's';
+                }
+
+                if (isset($this->input[$input_field_name]) && count($this->input[$input_field_name])) {
+                    $first_item = reset($this->input[$input_field_name]);
```

Comment:
Could you please add a comment in code to explain what are expected formats ? I had to read twice to be able to understand.